mutable_buffer/src/column.rs

```diff
@@ -41,6 +43,275 @@ pub enum Column {
 }
 
 impl Column {
+    /// Initializes a new column from typed values, the column on a table write
+    /// batach on an Entry. Will initialize the stats with the first
+    /// non-null value and update with any other non-null values included.
+    pub fn new_from_typed_values(
+        dictionary: &mut Dictionary,
+        row_count: usize,
+        logical_type: LogicalColumnType,
+        values: TypedValuesIterator<'_>,
+    ) -> Self {
+        match values {
+            TypedValuesIterator::String(vals) => match logical_type {
+                LogicalColumnType::Tag => {
+                    let mut tag_values = vec![None; row_count];
+                    let mut stats: Option<StatValues<String>> = None;
+
+                    for tag in vals {
+                        let tag_id = match tag {
+                            Some(tag) => {
+                                match stats.as_mut() {
+                                    Some(s) => StatValues::update_string(s, tag),
+                                    None => {
+                                        stats = Some(StatValues::new(tag.to_string()));
+                                    }
+                                }
+
+                                Some(dictionary.lookup_value_or_insert(tag))
+                            }
+                            None => None,
+                        };
```

Comment:
You could shorten this a little if you wanted to using `map`. The suggestion is correctly formatted for `rustfmt`

```suggestion
                        let tag_id = tag.map(|tag| {
                            if let Some(s) = stats.as_mut() {
                                StatValues::update_string(s, tag)
                            } else {
                                stats = Some(StatValues::new(tag.to_string()));
                            }
                            dictionary.lookup_value_or_insert(tag)
                        });
```